src/webhooks/index.ts

```diff
@@ -1,23 +1,29 @@
 import {ConfigInterface} from '../base-types';
 
+import {HttpWebhookRegistry} from './types';
 import {
-  addHttpHandler,
-  addHttpHandlers,
+  createAddHttpHandler,
+  createAddHttpHandlers,
+  createGetHttpHandler,
+  createGetTopicsAdded,
   createProcess,
   createRegister,
   createRegisterAllHttp,
-  getHttpHandler,
-  getTopicsAdded,
 } from './registry';
 
 export function shopifyWebhooks(config: ConfigInterface) {
+  const httpWebhookRegistry: HttpWebhookRegistry = {};
+
   return {
-    addHttpHandler,
-    addHttpHandlers,
-    getHttpHandler,
-    getTopicsAdded,
-    process: createProcess(config),
+    addHttpHandler: createAddHttpHandler(httpWebhookRegistry),
+    addHttpHandlers: createAddHttpHandlers(httpWebhookRegistry),
+    getHttpHandler: createGetHttpHandler(httpWebhookRegistry),
+    getTopicsAdded: createGetTopicsAdded(httpWebhookRegistry),
+    process: createProcess(config, httpWebhookRegistry),
     register: createRegister(config),
-    registerAllHttp: createRegisterAllHttp(config),
+    registerAllHttp: createRegisterAllHttp(config, httpWebhookRegistry),
+    testing: {
+      httpWebhookRegistry,
+    },
```

Comment:
Actually, there are only two checks needed against the structure...
- `'TOPIC' in registry`
- `Object.keys(registry)`

Rather than expose the registry itself under `testing`, I'll expose two methods that can return what's required for the `expect()` statements.

Reply:
Didn't we already have a method to return handled topics? It sounds to me like that's exactly what we need here.

Reply:
How would that happen? It seems natural to me for the test to run

```ts
expect(shopify.webhooks.getTopicsAdded()).toEqual([TOPIC_1, TOPIC_2]);
```

Or do I have the wrong end of the stick?